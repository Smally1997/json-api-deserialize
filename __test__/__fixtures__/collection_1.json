{
  "data": [
    {
      "id": "123",
      "type": "fundraiser",
      "attributes": {
        "updated_at": "2019-06-24T17:49:16.000Z",
        "created_at": "2019-05-09T15:28:44.000Z",
        "name": "Hogwarts Varsity Quidditch Team 2019"
      },
      "relationships": {
        "contact": {
          "data": {
            "id": "11650",
            "type": "contact"
          }
        },
        "organization": {
          "data": {
            "id": "19925",
            "type": "organization"
          }
        },
        "events": {
          "data": [
            {
              "id": "351978",
              "type": "event"
            },
            {
              "id": "351979",
              "type": "event"
            }
          ]
        }
      }
    },
    {
      "id": "1234",
      "type": "fundraiser",
      "attributes": {
        "updated_at": "2019-06-24T17:49:16.000Z",
        "created_at": "2019-05-09T15:28:44.000Z",
        "name": "Hogwarts JV Quidditch Team 2019"
      },
      "relationships": {
        "contact": {
          "data": {
            "id": "11651",
            "type": "contact"
          }
        },
        "organization": {
          "data": {
            "id": "19926",
            "type": "organization"
          }
        },
        "events": {
          "data": [
            {
              "id": "351980",
              "type": "event"
            },
            {
              "id": "351981",
              "type": "event"
            }
          ]
        }
      }
    }
  ],
  "included": [
    {
      "id": "27284",
      "type": "address",
      "attributes": {
        "updated_at": "2017-12-20T22:23:58.000Z",
        "created_at": "2017-12-20T22:23:58.000Z",
        "street": "123 Banffshire Ave",
        "city": "Dufftown",
        "region": "Scottish Highlands",
        "country": "Scotland",
        "zip": "12345",
        "abbreviated_region": "wa"
      }
    },
    {
      "id": "19925",
      "type": "organization",
      "attributes": {
        "updated_at": "2019-06-24T18:35:19.000Z",
        "created_at": "2018-11-20T21:41:49.000Z",
        "name": "Hogwarts Academy"
      },
      "relationships": {
        "address": {
          "data": {
            "id": "27284",
            "type": "address"
          }
        }
      }
    },
    {
      "id": "351978",
      "type": "event",
      "attributes": {
        "updated_at": "2019-06-21T18:30:25.000Z",
        "created_at": "2019-06-05T02:06:54.000Z",
        "name": "Training with Prof Snape",
        "status": "active"
      }
    },
    {
      "id": "351979",
      "type": "event",
      "attributes": {
        "updated_at": "2019-06-21T18:30:25.000Z",
        "created_at": "2019-06-05T02:06:54.000Z",
        "name": "Training with Prof Snape 2",
        "status": "pending"
      }
    },
    {
      "id": "27285",
      "type": "address",
      "attributes": {
        "updated_at": "2017-12-20T22:23:58.000Z",
        "created_at": "2017-12-20T22:23:58.000Z",
        "street": "124 Banffshire Ave",
        "city": "Dufftown 2",
        "region": "Scottish Highlands 2",
        "country": "Scotland 2",
        "zip": "12346",
        "abbreviated_region": "oh"
      }
    },
    {
      "id": "19926",
      "type": "organization",
      "attributes": {
        "updated_at": "2019-06-24T18:35:19.000Z",
        "created_at": "2018-11-20T21:41:49.000Z",
        "name": "Hogwarts Academy 2"
      },
      "relationships": {
        "address": {
          "data": {
            "id": "27285",
            "type": "address"
          }
        }
      }
    },
    {
      "id": "351980",
      "type": "event",
      "attributes": {
        "updated_at": "2019-06-21T18:30:25.000Z",
        "created_at": "2019-06-05T02:06:54.000Z",
        "name": "Training with Prof Snape 2",
        "status": "active"
      }
    },
    {
      "id": "351981",
      "type": "event",
      "attributes": {
        "updated_at": "2019-06-21T18:30:25.000Z",
        "created_at": "2019-06-05T02:06:54.000Z",
        "name": "Training with Prof Snape 3",
        "status": "pending"
      }
    }
  ]
}