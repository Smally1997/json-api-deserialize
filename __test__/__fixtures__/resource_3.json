{
  "data": {
    "type": "building",
    "id": "1",
    "attributes": {
      "name": "Mercer Island Books"
    },
    "relationships": {
      "owner": {
        "data": {
          "id": "2",
          "type": "owner"
        }
      }
    }
  },
  "included": [
    {
      "type": "owner",
      "id": "2",
      "attributes": {
        "name": "John Stamos"
      },
      "relationships": {
        "address": {
          "data": {
            "id": "3",
            "type": "address"
          }
        }
      }
    },
    {
      "type": "address",
      "id": "3",
      "attributes": {
        "street": "123 San Fran Baby Ave",
        "type": "commercial",
        "zip_code": "12345"
      },
      "relationships": {
        "driveway": {
          "data": {
            "id": "4",
            "type": "driveway"
          }
        }
      }
    },
    {
      "type": "driveway",
      "id": "4",
      "attributes": {
        "type": "The fancy circle type with a fountain in the middle"
      }
    }
  ]
}
