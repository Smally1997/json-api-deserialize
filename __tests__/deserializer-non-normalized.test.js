const fs = require('fs');
const path = require('path');

const deserialize = require('../dist').deserialize;

// Fixtures
const resourceResponses = [
  require('./__fixtures__/resource_1.json'),
  require('./__fixtures__/resource_2.json'),
  require('./__fixtures__/resource_3.json'),
  require('./__fixtures__/resource_4.json'),
  require('./__fixtures__/resource_5.json'),
  require('./__fixtures__/resource_6.json'),
  require('./__fixtures__/resource_7.json'),
  require('./__fixtures__/resource_8.json'),
  require('./__fixtures__/resource_9.json'),
];

const collectionResponses = [
  require('./__fixtures__/collection_1.json'),
  require('./__fixtures__/collection_2.json'),
];

describe('JsonApiDeserializer', () => {
  resourceResponses.forEach((resourceResponse, index) => {
    describe(`Resource Response (${index + 1})`, () => {
      const deserializedResponse = deserialize(resourceResponse, { normalize: false });

      fs.writeFileSync(
        path.join(__dirname, `./__outputs__/resource_non_normalized_${index + 1}.json`),
        JSON.stringify(deserializedResponse)
      );

      test('RO data object contains nested full resource objects', () => {
        expect(deserializedResponse).toMatchSnapshot();
      });

      test('RO shouldn\'t have included property', () => {
        expect(deserializedResponse.included).toBeFalsy();
      });
    });
  });

  collectionResponses.forEach((collectionResponse, index) => {
    describe(`Collection Response (${index + 1})`, () => {
      const deserializedResponse = deserialize(collectionResponse, { normalize: false });

      fs.writeFileSync(
        path.join(__dirname, `./__outputs__/collection_non_normalized_${index + 1}.json`),
        JSON.stringify(deserializedResponse)
      );

      test('RO data object contains nested full resource objects', () => {
        expect(deserializedResponse).toMatchSnapshot();
      });

      test('RO shouldn\'t have included property', () => {
        expect(deserializedResponse.included).toBeFalsy();
      });
    });
  });
});
