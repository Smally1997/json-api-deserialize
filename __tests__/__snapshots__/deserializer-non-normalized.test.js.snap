// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JsonApiDeserializer Collection Response (1) RO data object contains nested full resource objects 1`] = `
Object {
  "data": Array [
    Object {
      "attributes": Object {
        "created_at": "2019-05-09T15:28:44.000Z",
        "name": "Hogwarts Varsity Quidditch Team 2019",
        "updated_at": "2019-06-24T17:49:16.000Z",
      },
      "id": "123",
      "relationships": Object {
        "contact": Object {
          "data": Object {
            "id": "11650",
            "type": "contact",
          },
        },
        "events": Object {
          "data": Array [
            Object {
              "attributes": Object {
                "created_at": "2019-06-05T02:06:54.000Z",
                "name": "Training with Prof Snape",
                "status": "active",
                "updated_at": "2019-06-21T18:30:25.000Z",
              },
              "id": "351978",
              "type": "event",
            },
            Object {
              "attributes": Object {
                "created_at": "2019-06-05T02:06:54.000Z",
                "name": "Training with Prof Snape 2",
                "status": "pending",
                "updated_at": "2019-06-21T18:30:25.000Z",
              },
              "id": "351979",
              "type": "event",
            },
          ],
        },
        "organization": Object {
          "data": Object {
            "attributes": Object {
              "created_at": "2018-11-20T21:41:49.000Z",
              "name": "Hogwarts Academy",
              "updated_at": "2019-06-24T18:35:19.000Z",
            },
            "id": "19925",
            "relationships": Object {
              "address": Object {
                "data": Object {
                  "attributes": Object {
                    "abbreviated_region": "wa",
                    "city": "Dufftown",
                    "country": "Scotland",
                    "created_at": "2017-12-20T22:23:58.000Z",
                    "region": "Scottish Highlands",
                    "street": "123 Banffshire Ave",
                    "updated_at": "2017-12-20T22:23:58.000Z",
                    "zip": "12345",
                  },
                  "id": "27284",
                  "type": "address",
                },
              },
            },
            "type": "organization",
          },
        },
      },
      "type": "fundraiser",
    },
    Object {
      "attributes": Object {
        "created_at": "2019-05-09T15:28:44.000Z",
        "name": "Hogwarts JV Quidditch Team 2019",
        "updated_at": "2019-06-24T17:49:16.000Z",
      },
      "id": "1234",
      "relationships": Object {
        "contact": Object {
          "data": Object {
            "id": "11651",
            "type": "contact",
          },
        },
        "events": Object {
          "data": Array [
            Object {
              "attributes": Object {
                "created_at": "2019-06-05T02:06:54.000Z",
                "name": "Training with Prof Snape 2",
                "status": "active",
                "updated_at": "2019-06-21T18:30:25.000Z",
              },
              "id": "351980",
              "type": "event",
            },
            Object {
              "attributes": Object {
                "created_at": "2019-06-05T02:06:54.000Z",
                "name": "Training with Prof Snape 3",
                "status": "pending",
                "updated_at": "2019-06-21T18:30:25.000Z",
              },
              "id": "351981",
              "type": "event",
            },
          ],
        },
        "organization": Object {
          "data": Object {
            "attributes": Object {
              "created_at": "2018-11-20T21:41:49.000Z",
              "name": "Hogwarts Academy 2",
              "updated_at": "2019-06-24T18:35:19.000Z",
            },
            "id": "19926",
            "relationships": Object {
              "address": Object {
                "data": Object {
                  "attributes": Object {
                    "abbreviated_region": "oh",
                    "city": "Dufftown 2",
                    "country": "Scotland 2",
                    "created_at": "2017-12-20T22:23:58.000Z",
                    "region": "Scottish Highlands 2",
                    "street": "124 Banffshire Ave",
                    "updated_at": "2017-12-20T22:23:58.000Z",
                    "zip": "12346",
                  },
                  "id": "27285",
                  "type": "address",
                },
              },
            },
            "type": "organization",
          },
        },
      },
      "type": "fundraiser",
    },
  ],
  "deserialized": true,
}
`;

exports[`JsonApiDeserializer Collection Response (2) RO data object contains nested full resource objects 1`] = `
Object {
  "data": Array [
    Object {
      "attributes": Object {
        "title": "JSON:API paints my bikeshed!",
      },
      "id": "1",
      "links": Object {
        "self": "http://example.com/articles/1",
      },
      "relationships": Object {
        "author": Object {
          "data": Object {
            "attributes": Object {
              "first-name": "Dan",
              "last-name": "Gebhardt",
              "twitter": "dgeb",
              "type": "human",
            },
            "id": "9",
            "links": Object {
              "self": "http://example.com/people/9",
            },
            "type": "people",
          },
          "links": Object {
            "related": "http://example.com/articles/1/author",
            "self": "http://example.com/articles/1/relationships/author",
          },
        },
        "comments": Object {
          "data": Array [
            Object {
              "attributes": Object {
                "body": "First!",
              },
              "id": "5",
              "links": Object {
                "self": "http://example.com/comments/5",
              },
              "relationships": Object {
                "author": Object {
                  "data": Object {
                    "id": "2",
                    "type": "people",
                  },
                },
              },
              "type": "comments",
            },
            Object {
              "attributes": Object {
                "body": "I like XML better",
              },
              "id": "12",
              "links": Object {
                "self": "http://example.com/comments/12",
              },
              "relationships": Object {
                "author": Object {
                  "data": Object {
                    "attributes": Object {
                      "first-name": "Dan",
                      "last-name": "Gebhardt",
                      "twitter": "dgeb",
                      "type": "human",
                    },
                    "id": "9",
                    "links": Object {
                      "self": "http://example.com/people/9",
                    },
                    "type": "people",
                  },
                },
              },
              "type": "comments",
            },
          ],
          "links": Object {
            "related": "http://example.com/articles/1/comments",
            "self": "http://example.com/articles/1/relationships/comments",
          },
        },
      },
      "type": "articles",
    },
  ],
  "deserialized": true,
}
`;

exports[`JsonApiDeserializer Resource Response (1) RO data object contains nested full resource objects 1`] = `
Object {
  "data": Object {
    "attributes": Object {
      "created_at": "2019-05-09T15:28:44.000Z",
      "name": "Hogwarts Varsity Quidditch Team 2019",
      "updated_at": "2019-06-24T17:49:16.000Z",
    },
    "id": "123",
    "relationships": Object {
      "contact": Object {
        "data": Object {
          "id": "11650",
          "type": "contact",
        },
      },
      "events": Object {
        "data": Array [
          Object {
            "attributes": Object {
              "created_at": "2019-06-05T02:06:54.000Z",
              "name": "Training with Prof Snape",
              "status": "active",
              "updated_at": "2019-06-21T18:30:25.000Z",
            },
            "id": "351978",
            "type": "event",
          },
          Object {
            "attributes": Object {
              "created_at": "2019-06-05T02:06:54.000Z",
              "name": "Training with Prof Snape 2",
              "status": "pending",
              "updated_at": "2019-06-21T18:30:25.000Z",
            },
            "id": "351979",
            "type": "event",
          },
        ],
      },
      "organization": Object {
        "data": Object {
          "attributes": Object {
            "created_at": "2018-11-20T21:41:49.000Z",
            "name": "Hogwarts Academy",
            "updated_at": "2019-06-24T18:35:19.000Z",
          },
          "id": "19925",
          "relationships": Object {
            "address": Object {
              "data": Object {
                "attributes": Object {
                  "abbreviated_region": "wa",
                  "city": "Dufftown",
                  "country": "Scotland",
                  "created_at": "2017-12-20T22:23:58.000Z",
                  "region": "Scottish Highlands",
                  "street": "123 Banffshire Ave",
                  "updated_at": "2017-12-20T22:23:58.000Z",
                  "zip": "12345",
                },
                "id": "27284",
                "type": "address",
              },
            },
          },
          "type": "organization",
        },
      },
    },
    "type": "fundraiser",
  },
  "deserialized": true,
}
`;

exports[`JsonApiDeserializer Resource Response (2) RO data object contains nested full resource objects 1`] = `
Object {
  "data": Object {
    "attributes": Object {
      "body": "The shortest article. Ever.",
      "created_at": "2015-05-22T14:56:29.000Z",
      "title": "JSON:API paints my bikeshed!",
      "updated_at": "2015-05-22T14:56:28.000Z",
    },
    "id": "1",
    "relationships": Object {
      "author": Object {
        "data": Object {
          "attributes": Object {
            "age": 80,
            "gender": "male",
            "name": "John",
          },
          "id": "42",
          "type": "people",
        },
      },
    },
    "type": "articles",
  },
  "deserialized": true,
}
`;

exports[`JsonApiDeserializer Resource Response (3) RO data object contains nested full resource objects 1`] = `
Object {
  "data": Object {
    "attributes": Object {
      "created_at": "2019-06-30T22:33:41+0000",
      "name": "Mercer Island Book Store",
      "updated_at": "2019-06-30T22:33:41+0000",
    },
    "id": "1",
    "relationships": Object {
      "owner": Object {
        "data": Object {
          "attributes": Object {
            "created_at": "2019-06-30T22:33:41+0000",
            "full_name": "John Doe",
            "updated_at": "2019-06-30T22:33:41+0000",
          },
          "id": "2",
          "relationships": Object {
            "address": Object {
              "data": Object {
                "attributes": Object {
                  "created_at": "2019-06-30T22:33:41+0000",
                  "street": "123 Test Ave",
                  "type": "commercial",
                  "updated_at": "2019-06-30T22:33:41+0000",
                  "zip_code": "12345",
                },
                "id": "3",
                "relationships": Object {
                  "driveway": Object {
                    "data": Object {
                      "attributes": Object {
                        "created_at": "2019-06-30T22:33:41+0000",
                        "type": "parking_lot",
                        "updated_at": "2019-06-30T22:33:41+0000",
                      },
                      "id": "4",
                      "type": "driveway",
                    },
                  },
                },
                "type": "address",
              },
            },
          },
          "type": "owner",
        },
      },
    },
    "type": "book",
  },
  "deserialized": true,
}
`;

exports[`JsonApiDeserializer Resource Response (4) RO data object contains nested full resource objects 1`] = `
Object {
  "data": Object {
    "attributes": Object {
      "body": "The shortest article. Ever.",
      "created_at": "2015-05-22T14:56:29.000Z",
      "title": "JSON:API paints my bikeshed!",
      "updated_at": "2015-05-22T14:56:28.000Z",
    },
    "id": "1",
    "relationships": Object {
      "author": Object {
        "data": Object {
          "id": "42",
          "type": "people",
        },
      },
    },
    "type": "articles",
  },
  "deserialized": true,
  "meta": Object {
    "foo": "bar",
  },
}
`;
